src/main/kotlin/com/cognifide/gradle/aem/pkg/DownloadTask.kt

```diff
@@ -0,0 +1,141 @@
+package com.cognifide.gradle.aem.pkg
+
+import com.cognifide.gradle.aem.api.AemConfig
+import com.cognifide.gradle.aem.api.AemTask
+import com.cognifide.gradle.aem.base.vlt.CheckoutConfig
+import com.cognifide.gradle.aem.instance.Instance
+import com.cognifide.gradle.aem.instance.InstanceSync
+import com.cognifide.gradle.aem.internal.PropertyParser
+import com.cognifide.gradle.aem.internal.file.FileContentReader
+import com.cognifide.gradle.aem.internal.file.FileException
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.api.tasks.*
+import org.gradle.api.tasks.bundling.Zip
+import java.io.File
+
+open class DownloadTask : Zip(), AemTask {
+
+    companion object {
+        const val NAME = "aemDownload"
+        const val CLASSIFIER_DOWNLOAD = "download"
+        const val CLASSIFIER_SHELL = "shell"
+        const val EXTRACT_FLAG = "aem.download.extract"
+        const val FORCE_NEW_FLAG = "aem.download.extract.force.new"
+    }
+
+    @Nested
+    final override val config = AemConfig.of(project)
+
+    @InputDirectory
+    private val prepareDir = AemTask.taskDir(project, PrepareTask.NAME)
+
+    @Internal
+    val props = PropertyParser(project)
+
+    @Input
+    val forceNewFlag = props.flag(FORCE_NEW_FLAG)
+
+    @Input
+    val extractFlag = props.flag(EXTRACT_FLAG)
```

Comment:
inline const because below you have no constants. "no matter how as long as consistent" :> / my own fav idiom says.